lib/internal/test_runner/test.js

```diff
@@ -517,12 +524,10 @@ class Test extends AsyncResource {
         return;
       }
 
-      if (this.parent?.hooks.afterEach.length > 0) {
-        await this.parent[kRunHook]('afterEach', { args, ctx });
-      }
-
+      await afterEach();
       this.pass();
     } catch (err) {
+      await afterEach();
```

Comment:
Do we have a test where the test throws and the `afterEach()` also throws? If not, can we add one?

Reply:
added a test and a fix for this case, thanks for the catch!